javatests/arcs/android/entity/TtlHandleTest.kt

```diff
@@ -0,0 +1,202 @@
+package arcs.android.entity
+
+import androidx.test.core.app.ApplicationProvider
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import arcs.android.storage.database.AndroidSqliteDatabaseManager
+import arcs.core.data.HandleMode
+import arcs.core.data.Ttl
+import arcs.core.entity.DummyEntity
+import arcs.core.entity.HandleContainerType
+import arcs.core.entity.HandleSpec
+import arcs.core.entity.ReadWriteCollectionHandle
+import arcs.core.entity.ReadWriteSingletonHandle
+import arcs.core.entity.SchemaRegistry
+import arcs.core.host.EntityHandleManager
+import arcs.core.storage.StorageKey
+import arcs.core.storage.api.DriverAndKeyConfigurator
+import arcs.core.storage.keys.DatabaseStorageKey
+import arcs.core.storage.referencemode.ReferenceModeStorageKey
+import arcs.core.util.Time
+import arcs.jvm.util.testutil.FakeTime
+import com.google.common.truth.Truth.assertThat
+import kotlinx.coroutines.test.runBlockingTest
+import org.junit.Before
+import org.junit.Test
+import org.junit.runner.RunWith
+
+@Suppress("EXPERIMENTAL_API_USAGE")
+@RunWith(AndroidJUnit4::class)
+class TtlHandleTest() {
+
+    private val backingKey = DatabaseStorageKey.Persistent("entities-backing", DummyEntity.SCHEMA_HASH)
+    private val collectionKey = ReferenceModeStorageKey(
+        backingKey = backingKey,
+        storageKey = DatabaseStorageKey.Persistent("collection", DummyEntity.SCHEMA_HASH)
+    )
+    private val singletonKey = ReferenceModeStorageKey(
+        backingKey = backingKey,
+        storageKey = DatabaseStorageKey.Persistent("singleton", DummyEntity.SCHEMA_HASH)
+    )
+    private lateinit var databaseManager: AndroidSqliteDatabaseManager
+    
+    @Before
+    fun setUp() {
+        databaseManager = AndroidSqliteDatabaseManager(ApplicationProvider.getApplicationContext())
+        DriverAndKeyConfigurator.configure(databaseManager)
+        SchemaRegistry.register(DummyEntity)
+    }
+
+    @Test
+    fun singletonWithExpiredEntities() = runBlockingTest {
+        val time = FakeTime()
```

Comment:
Can this be created as a property of the test class, as part of `setUp`? Then the helper methods can just use the time field directly, and there's no need to pass it to the handle creation helpers.

Reply:
Done